static/search.js

```diff
@@ -184,48 +188,50 @@ var Search = Control.extend({
 	//
 	// @param dataUrl the url of the docMapHash.json file
 	//
-	// @returns promise that resolves to true if localStorage was cleared and false otherwise
+	// @returns thenable that resolves to true if localStorage was cleared and false otherwise
 	checkDocMapHash(dataUrl) {
-		
-		return new Promise((resolve, reject) => {
-				//no need to do anything if localStorage isn't present
-				if(!window.localStorage){
-					resolve(false);
-					return;
-				}
+		var self = this,
+				returnDeferred = $.Deferred();
 
-				$.ajax({
-					url: dataUrl,
-					dataType: "json",
-					cache: false
-				}).then((data) => {
-					let localStorageKey = this.formatLocalStorageKey(this.docMapHashLocalStorageKey),
-							docMapHashLocalStorage = this.getLocalStorageItem(localStorageKey),
-							lsHash = docMapHashLocalStorage && docMapHashLocalStorage.hash,
-							dataHash = data && data.hash;
-
-					//no lsHash && no dataHash => resolve
-					//lsHash && no dataHash => resolve
-					if(!dataHash){
-						resolve(false);
-						return;
-					}
+		//no need to do anything if localStorage isn't present
+		if(!window.localStorage){
+			returnDeferred.resolve(false);
+			return;
+		}
 
-					//no lsHash && dataHash => save && resolve
-					//lsHash && dataHash => check if same
-					if(lsHash !== dataHash){
+		$.ajax({
+			url: dataUrl,
+			dataType: "json",
+			cache: false
+		}).then(function(data){
+			var localStorageKey = self.formatLocalStorageKey(self.docMapHashLocalStorageKey),
+					docMapHashLocalStorage = self.getLocalStorageItem(localStorageKey),
+					lsHash = docMapHashLocalStorage && docMapHashLocalStorage.hash,
+					dataHash = data && data.hash;
+
+			//no lsHash && no dataHash => resolve
+			//lsHash && no dataHash => resolve
+			if(!dataHash){
+				returnDeferred.resolve(false);
+				return;
+			}
 
-						//TODO: wait until after we've attempted to get a new
-						//searchMap before clearing?
-						localStorage.clear();
-						this.setLocalStorageItem(localStorageKey, data);
-						resolve(true);
-						return;
-					}
+			//no lsHash && dataHash => save && resolve
+			//lsHash && dataHash => check if same
+			if(lsHash !== dataHash){
 
-					resolve(false);
-				}, reject);
-		});
+				//TODO: wait until after we've attempted to get a new
+				//searchMap before clearing?
```

Comment:
This seems like a good idea. Will this be implemented before we deploy the search feature?